src/components/NcModal/NcModal.vue

```diff
@@ -1215,5 +871,160 @@ $pi: 3.14159265358979;
 		opacity: 1;
 	}
 }
+</style>
+
+<docs>
+The `NcModel` is the base component used for modals and dialogs.
+While `NcDialog` should be used for general dialogs like confirmations or forms,
+`NcModal` allows for custom content like showing image multimedia.
+
+For showing the modal you can use either `v-model:show="showModal"` or `v-if` on the `NcModal`,
+depending on whether you require the Modal to stay within the DOM or not. Do not mix both, as this will break the out transition animation.
 
+```vue
+<template>
+	<div>
+		<NcButton @click="showModal">Show Modal</NcButton>
+		<NcModal
+			v-model:show="modal"
+			@close="closeModal"
+			size="small"
+			name="Name"
+			out-transition>
+			<template #actions>
+				<NcActionCaption name="Some action" />
+			</template>
+			<div class="modal__content">Hello world</div>
+		</NcModal>
+	</div>
+</template>
+<script>
+export default {
+	data() {
+		return {
+			modal: false
+		}
+	},
+	methods: {
+		showModal() {
+			this.modal = true
+		},
+		closeModal() {
+			this.modal = false
+		}
+	}
+}
+</script>
+<style scoped>
+.modal__content {
+	margin: 50px;
+	text-align: center;
+}
+</style>
+```
+
+### Modal with slideshow
+
+```vue
+<template>
+	<div>
+		<NcButton @click="isOpen = true">Show Modal</NcButton>
+		<NcModal
+			v-if="isOpen"
+			close-button-outside
+			enable-slideshow
+			:has-next="page < lastPage"
+			:has-previous="page > 0"
+			name="Modal with slideshow"
+			@next="page++"
+			@previous="page--"
+			@close="isOpen = false">
+			<div class="modal__content" :style="{ background: currentPage.background }">
+				<p class="model__content-text">{{ currentPage.text }}</p>
+			</div>
+		</NcModal>
+	</div>
+</template>
+<script>
+const PAGES = [
+	{ text: 'First page', background: 'linear-gradient(#e66465, #9198e5)' },
+	{ text: 'Second page', background: 'linear-gradient(0.25turn, #3f87a6, #ebf8e1, #f69d3c)' },
+	{ text: 'Third page', background: 'lightblue' },
+	{ text: 'Last page', background: 'lightgrey' },
+]
+
+export default {
+	data() {
+		return {
+			isOpen: false,
+			page: 0,
+			lastPage: PAGES.length - 1,
+		}
+	},
+	computed: {
+		currentPage() {
+			return PAGES[this.page]
+		},
+	}
+}
+</script>
+<style scoped>
+.modal__content {
+	height: 100%;
+	min-height: 30vh;
+	display: flex;
+	align-items: center;
+	justify-content: center;
+}
+
+.model__content-text {
+	font-size: 16px;
+	font-weight: bold;
+}
 </style>
+```
+
+### Usage of popover in modal
+
+* Set container property to .modal-mask to inject popover context of the modal:
+
+```vue
+<template>
+	<div>
+		<NcButton @click="showModal">Show Modal</NcButton>
+		<NcModal v-if="modal" @close="closeModal" size="small" class="emoji-modal">
+			<NcEmojiPicker container=".emoji-modal" @select="select">
```

Comment:
Example is no longer working, caused by teleport? 

<img width="1042" height="212" alt="Image" src="https://github.com/user-attachments/assets/c492e696-9f2e-41d6-9c14-14fdc11c5136" />